plugins/inputs/system/system.go

```diff
@@ -22,54 +22,81 @@ import (
 var sampleConfig string
 
 type SystemStats struct {
+	// Publishes all metrics at once, opposed to three separate metrics
+	// Enabling this will lose metric type support (used in outputs such as prometheus)
+	MergeMetrics bool `toml:"merge_metrics"`
+
 	Log telegraf.Logger
 }
 
 func (*SystemStats) SampleConfig() string {
 	return sampleConfig
 }
 
-func (s *SystemStats) Gather(acc telegraf.Accumulator) error {
-	loadavg, err := load.Avg()
-	if err != nil && !strings.Contains(err.Error(), "not implemented") {
-		return err
+func (s *SystemStats) addFields(acc telegraf.Accumulator, loadFields map[string]interface{}, uptimeFields map[string]interface{}, uptimeFormatFields map[string]interface{}) {
+	now := time.Now()
+
+	if !s.MergeMetrics {
+		acc.AddGauge("system", loadFields, nil, now)
+		acc.AddCounter("system", uptimeFields, nil, now)
+		acc.AddFields("system", uptimeFormatFields, nil, now)
+		return
 	}
 
-	numCPUs, err := cpu.Counts(true)
-	if err != nil {
-		return err
+	mergedFields := map[string]interface{}{}
+
+	for k, v := range loadFields {
+		mergedFields[k] = v
+	}
+	for k, v := range uptimeFields {
+		mergedFields[k] = v
 	}
+	for k, v := range uptimeFormatFields {
+		mergedFields[k] = v
+	}
+
+	acc.AddFields("system", mergedFields, nil, now)
+}
 
-	fields := map[string]interface{}{
-		"load1":  loadavg.Load1,
-		"load5":  loadavg.Load5,
-		"load15": loadavg.Load15,
-		"n_cpus": numCPUs,
+func (s *SystemStats) Gather(acc telegraf.Accumulator) error {
+	loadavg, err := load.Avg()
+	if err != nil && !strings.Contains(err.Error(), "not implemented") {
+		return err
 	}
 
+	loadFields := map[string]interface{}{}
 	users, err := host.Users()
 	if err == nil {
-		fields["n_users"] = len(users)
+		loadFields["n_users"] = len(users)
 	} else if os.IsNotExist(err) {
 		s.Log.Debugf("Reading users: %s", err.Error())
 	} else if os.IsPermission(err) {
 		s.Log.Debug(err.Error())
 	}
 
-	now := time.Now()
-	acc.AddGauge("system", fields, nil, now)
+	numCPUs, err := cpu.Counts(true)
+	if err != nil {
+		return err
+	}
+
+	loadFields["load1"] = loadavg.Load1
+	loadFields["load5"] = loadavg.Load5
+	loadFields["load15"] = loadavg.Load15
```

Comment:
It may happen that `loadavg` is `nil` here. 

Maybe we can skip adding fields from `loadavg` struct for that case?